### PR TITLE
Cleanup sig-instrumentation teams

### PR DESCRIPTION
**Pull request by dgrisonnet**
/cc @logicalhan @ehashman @dashpole

### CODE REVIEW

File: config/kubernetes/sig-instrumentation/teams.yaml [hunk 1]
```diff
@@ -59,3 +59,31 @@ teams:
...
+  klog-maintainers:
+    description: Write access to the klog repo
+    members:
+    - dims
+    - erictune
+    - justinsb
+    - lavalamp
+    - tallclair
+    - thockin
+    privacy: closed
```
Comment (dgrisonnet) [inline config/kubernetes/sig-instrumentation/teams.yaml:77]: We most likely want to update these teams, any thoughts and who should be there? I would at least expect WG-structured-logging leads

Comment (dashpole) [reply config/kubernetes/sig-instrumentation/teams.yaml:77]: maybe mirror the OWNERS file? https://github.com/kubernetes/klog/blob/main/OWNERS

Comment (dgrisonnet) [reply config/kubernetes/sig-instrumentation/teams.yaml]: :+1: I updated the teams to only include klog approvers.

File: config/kubernetes/sig-instrumentation/teams.yaml [hunk 2]
```diff
@@ -59,3 +59,29 @@ teams:
...
+    - serathius
+    - thockin
+    privacy: closed
+  klog-maintainers:
+    description: Write access to the klog repo
+    members:
+    - dims
+    - serathius
+    - thockin
+    privacy: closed
```
[the remark this answers is not in the document]
Comment (serathius) [reply config/kubernetes/sig-instrumentation/teams.yaml:75]: Looks good.

File: config/kubernetes/sig-instrumentation/teams.yaml (hunk 1 above)
Comment (serathius) [reply config/kubernetes/sig-instrumentation/teams.yaml]: I think we should update klog/OWNERS I was surprised not to find @pohly there.

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**